test/accountantPunishment.js

```diff
@@ -0,0 +1,238 @@
+        await registry.registerIdentity(accountant.address, channelStake, Zero, beneficiaries[0], signature)
+        expect(await registry.isRegistered(provider.address)).to.be.true
+        expect(await accountant.isOpened(expectedChannelId)).to.be.true
+
+        // Channel stake have to be transfered to accountant
+        const accountantTokenBalance = await token.balanceOf(accountant.address)
+        accountantTokenBalance.should.be.bignumber.equal(initialAccountantBalance.add(channelStake))
+
+        const channel = await accountant.channels(expectedChannelId)
+        expect(channel.balance.toNumber()).to.be.equal(channelStake.toNumber())
+
+        // Accountant should still not have available balance
+        const availableBalance = await accountant.availableBalance()
+        availableBalance.should.be.bignumber.equal(Zero)
+    })
+
+    it('should settle promise and decrease channel balance', async () => {
+        const channelId = generateChannelId(provider.address, accountant.address)
+        const amount = new BN(250)
+        const R = randomBytes(32)
+        const hashlock = keccak(R)
+
+        // Create accountant promise
+        const promise = createPromise(channelId, amount, Zero, hashlock, accountantOperator)
+
+        // Settle promise
+        const initialChannelBalance = (await accountant.channels(channelId)).balance
+        const expectedChannelBalance = initialChannelBalance.sub(amount)
+
+        await accountant.settlePromise(promise.channelId, promise.amount, promise.fee, R, promise.signature)
+
+        const channelBalance = (await accountant.channels(channelId)).balance
+        channelBalance.should.be.bignumber.equal(expectedChannelBalance)
+    })
+
+    it('should rebalance channel only with available abalance and enable punishment mode', async () => {
+        const channelId = generateChannelId(provider.address, accountant.address)
+        const channel = await accountant.channels(channelId)
+        const rebalanceAmount = channel.loan.sub(channel.balance)
+        const initialStake = await accountant.getStake()
+
+        // Make accountant available balance to be half of needed
+        await topUpTokens(token, accountant.address, rebalanceAmount / 2)
+
+        // Rebalance channel
+        await accountant.rebalanceChannel(channelId)
+
+        // Stake should remain untouched
+        const accountantStake = await accountant.getStake()
+        expect(accountantStake.toNumber()).to.be.equal(initialStake.toNumber())
+
+        // There should be zoro available balance
+        expect((await accountant.availableBalance()).toNumber()).to.be.equal(0)
+
+        // Because of not getting all expected balance, there should be enabled punishment mode
+        const accountantStatus = await accountant.getStatus() // 0 - Active, 1 - Paused, 2 - Punishment, 3 - Closed
+        expect(accountantStatus.toNumber()).to.be.equal(2)
+        expect(await accountant.isAccountantActive()).to.be.false
```

Comment:
But accountant is still active, just he is in punishment mode. Or its just reading as a false which is '0' - active?

Reply:
`isAccountantActive()` will say false. Otherwise this test will fail. accountantStatus = 2 means punishment mode.

So when accountant is in punishment mode (status = 2) then he is not active (isAccountantActive = false).